lib/parallel_tests/test/runner.rb

```diff
@@ -85,6 +85,39 @@ def tests_with_size(tests, options)
           tests
         end
 
+        def process_in_batches(cmd, os_cmd_length_limit, tests)
+          # Filter elements not starting with value in tests to retain in each batch
+          # i.e. retain common parameters for each batch
+          retained_elements = cmd.reject { |s| s.start_with?(tests) }
+        
+          # elements that needs to be checked for length and sliced into batches
+          non_retained_elements = cmd.select { |s| s.start_with?(tests) }
```

Comment:
use `.partition` instead of select/reject

Reply:
modified to use partition